extra-cpp-bindings/src/walletconnect.rs

```diff
@@ -473,7 +495,84 @@ impl WalletconnectClient {
         let signed_tx = &typedtx.rlp_signed(&sig);
         Ok(signed_tx.to_vec())
     }
-    pub fn erc20_transfer(&mut self, info: &WalletConnectErc20Transfer) -> Result<Vec<u8>> {
+
+    fn get_sent_tx_raw_bytes(
+        &self,
+        newclient: Client,
+        signeraddress: H160,
+        typedtx: &mut TypedTransaction,
+        common: &WalletConnectTxCommon,
+    ) -> Result<Vec<u8>> {
+        let mynonce = U256::from_dec_str(&common.nonce)?;
+        if !mynonce.is_zero() {
+            typedtx.set_nonce(mynonce);
+        }
+        typedtx.set_from(signeraddress);
+        if !common.chainid == 0 {
+            typedtx.set_chain_id(common.chainid);
+        }
+        if !common.gas_limit.is_empty() {
+            typedtx.set_gas(U256::from_dec_str(&common.gas_limit)?);
+        }
+        if !common.gas_price.is_empty() {
+            typedtx.set_gas_price(U256::from_dec_str(&common.gas_price)?);
+        }
+
+        let tx_bytes = self
+            .rt
+            .block_on(send_typed_tx(newclient, typedtx.clone(), signeraddress))
+            .map_err(|e| anyhow!("send_typed_transaction error {}", e.to_string()))?;
+
+        Ok(tx_bytes.0.to_vec())
+    }
+
+    pub fn erc20_transfer(
```

Comment:
duplicating all these methods on a different client (walletconnect) doesn't feel right, but I assume bindings can't handle generics? Anyway, maybe worth exploring adding a pure C++ wrapper on top of bindings that will be generic over the client, so that the end user can just use the same code for local wallet / walletconnect1 / walletconnect2 by just plugging a different client in the wrapper's constructor (and no other change is needed)

Reply:
Indeed, erc20_transfer, erc20_transfer_from and others are really duplicated from the local wallet. And may duplicate in Walletconnect 2.0 in the future. This PR just fix the bugs that those functions can not be transferred by metamask/defiwallet. Considering generic those functions are not a easy task, and the unreal part is also needed to update, it is worth to do it in another [PR](https://github.com/cronos-labs/play-cpp-sdk/issues/431).

Reply:
Support metamask/defiwallet so that users can use them to send transactions first (since it is critical that many users are complaining that can not use metamask to interact with contracts), then refactor would be better.